engine/execution/computation/manager.go

```diff
@@ -53,10 +63,20 @@ type ComputationManager interface {
 	GetAccount(addr flow.Address, header *flow.Header, view state.View) (*flow.Account, error)
 }
 
-var DefaultScriptLogThreshold = 1 * time.Second
-var DefaultScriptExecutionTimeLimit = 10 * time.Second
-
-const MaxScriptErrorMessageSize = 1000 // 1000 chars
+type ComputationConfig struct {
+	CadenceTracing           bool
+	ExtensiveTracing         bool
+	ProgramsCacheSize        uint
+	ScriptLogThreshold       time.Duration
+	ScriptExecutionTimeLimit time.Duration
+
+	// When NewCustomVirtualMachine is nil, the manager will create a standard
+	// fvm virtual machine via fvm.NewVirtualMachine.  Otherwise, the manager
+	// will create a virtual machine using this function.
+	//
+	// Note that this is primarily used for testing.
```

Comment:
Maybe we could create a separate internal function `newWithVirtualMachine`. So that `New` will create a VM using the standard VM, and for test case, which also has access to `newWithVirtualMachine` will create a different VM  for testing purpose

Reply:
The tests completely stub out the vm and has nothing in common with the real vm (see manager_test.go in this PR).  I'm not sure if newWithVirtualMachine will be useful.